Bazel clean up

brocollie08 opened the pull request:
<!-- 

Describe what's changing, why, and any other background info.

Make sure to add:
  - Tests
  - Documentation Updates

-->

https://github.com/player-ui/player/issues/424


### Change Type (required)
Indicate the type of change your pull request is:

<!-- 
  We use semantic versioning: https://semver.org/. Review that documentation for 
  more detailed guidelines.
-->
- [x] `patch`
- [ ] `minor`
- [ ] `major`

<!--
  To include release notes in the automatic changelong, just add a level 1 markdown header below
  and include any markdown notes to go into the changelog: https://intuit.github.io/auto/docs/generated/changelog#additional-release-notes

  Example:

  # Release Notes
  Added new plugin, to use it:
  ```typescript
  const plugin = new Plugin(...)
  ```
-->


### Does your PR have any documentation updates?
- [ ] Updated docs
- [x] No Update needed
- [ ] Unable to update docs
<!--
In an effort to standardize our process and code, please make sure you include documentation and/or update any existing documentation.
Please refer to our site https://player-ui.github.io/latest/about, and include any neccesary information that would be helpful to coders, developers, and learners.

If you are unable to update the current documents, please create an issue for us to get back to it.

-->

sugarmanz commented on jvm/core/src/main/kotlin/com/intuit/playerui/core/plugins/JSScriptPluginWrapper.kt:
```diff
@@ -18,15 +18,10 @@ public abstract class JSScriptPluginWrapper(public val name: String, protected v
 
     final override lateinit var instance: Node protected set
 
-    protected val debugScript: String
-        get() = loadDebugScript() ?: script
-
-    private fun loadDebugScript(): String? = JSScriptPluginWrapper::class.java.classLoader.getResource(sourcePath?.substringBeforeLast(".") + ".dev." + sourcePath?.substringAfterLast("."))?.readText()
-
     public val isInstantiated: Boolean get() = ::instance.isInitialized
 
     override fun apply(runtime: Runtime<*>) {
-        runtime.load(ScriptContext(if (runtime.config.debuggable) debugScript else script, sourcePath ?: "$name.js"))
```
Can you explain what the forward looking strategy is here since we're technically removing some functionality?

brocollie08 replied:
two things here:

1. the functionality isn't working 😄 
2. once it does work, I think we'll remove the config option since it'll always be "debuggable" as long as you're pulling in the correct j2v8 dep, reg vs debuggable

brocollie08 commented on android/player/src/main/java/com/intuit/playerui/android/asset/SuspendableAsset.kt:
```diff
@@ -153,6 +152,7 @@ public abstract class SuspendableAsset<Data>(assetContext: AssetContext, seriali
                 }
             }
             hydratedView.await().also { parent ->
+                replaceSelfWithView(parent)
```
we were able to fix the test w/o this, by explicitly calling into `shouldBeView` for nested components. This change isn't 100% necessary but doing so made it possible to revert the `shouldBeView` calls, one at the top is enough.

that's a good callout though, we basically want this behaviour for tests only

sugarmanz replied:
> that's a good callout though, we basically want this behaviour for tests only

Ooo, so in the tests, can we trigger `onViewAttachedToWindow` in the `onView` handler here before we emit to the view channel:
https://github.com/player-ui/player/blob/5ae4ae089596add787ddfb855431c41cc463f5c6/plugins/reference-assets/android/src/androidTest/java/com/intuit/playerui/android/reference/assets/test/AssetTest.kt#L86

brocollie08 commented:
To summarize the databinding/viewbinding changes, I decided to rip it out wholesale instead of leaving in any remnants or references to the db dependency.

The `enable_data_binding` flag was generating the databinding classes based on the resources wrapped in `<layout>` tag, which was actually fine when it came to building the demo app and running it, but this caused an issue with the distribution aar (`player_base`) complaining about those generated classes not being able to resolve.

removing that flag solved that issue, but created another problem at run time where the app would crash right away complaining about the presence of the `<layout>` tags without the enabling of db -> leading to the removal of the tags

at this point i've removed most of the db related code, and IIRC the imports in the manually created "binding" classes were no longer resolving, so I removed those as well. That class has been called out to not be generated by ViewBinding, it's just a placeholder class with the existing API to make it look like it's generated. I'm not sure if it's fair to expect it to work like a fully functional generated binding class.

all of this together made me think it's better to get rid of all of it, instead of having bits and pieces making it look like it's in there. This actually reminds me, i should remove the deps as well

brocollie08 commented on WORKSPACE.bzlmod:
```diff
@@ -9,22 +9,11 @@ android_sdk_repository(
 bind(
     name = "databinding_annotation_processor",
     actual = "//:compiler_annotation_processor",
 )
```
I think these are actually required by bazel android tools 😬 
```
target 'databinding_annotation_processor' not declared in package 'external' defined by /Users/slin8/Work/player/WORKSPACE (Tip: use `query "//external:*"` to see all the targets in that package) and referenced by '@bazel_tools//tools/android:databinding_annotation_processor'
```

brocollie08 replied:
this seems really weird though..

sugarmanz replied:
Yeah, this doesn't seem like something we should need to do at all. But if it's breaking other things, we can leave for future cleanup

sugarmanz commented:
> but this caused an issue with the distribution aar (player_base) complaining about those generated classes not being able to resolve

Could this just be fixed by fixing [this](https://github.com/player-ui/player/pull/435#discussion_r1681645220)?

> I decided to rip it out wholesale instead of leaving in any remnants or references to the db dependency

My main concern with this is it's a major change to the components we currently expose for consumers wrt the `PlayerFragment` integration. If there is usage of the view binding components in a generic view binding pattern (i.e. referencing `ViewBinding` base class), there isn't really a great workaround here.

> I'm not sure if it's fair to expect it to work like a fully functional generated binding class.

To extend this, if we're removing view binding support altogether, then we should address how the classes are currently structured/used to not pretend it's a view binding (i.e. just use `findViewById` or some other mechanism).

But this all goes back to whether we want to enable view binding support in the future. It's conflicting to put more work in to get further away from our target state.

sugarmanz commented on android/player/BUILD:
```diff
@@ -1,33 +1,43 @@
 
-distribution(
+android_library(
     name = "player",
-    lib_name = "player-databinding",
+    custom_package = "com.intuit.playerui.android",
+    resource_files = glob(["src/main/res/**"]),
+    manifest = ":src/main/AndroidManifest.xml",
+    tags = ["maven_coordinates=com.intuit.playerui:android:aar:%s" % VERSION],
+    deps = main_deps,
+    exports = main_exports + [":player_android"],
+    visibility = ["//visibility:public"],
+)
```
We could potentially remove the need for this layer by patching `rules_kotlin` to add the manifest to the top-level `android_library` returned from `kt_android_library` - can be done another time, just wanted to comment

brocollie08 replied:
great first issue

sugarmanz commented on android/player/BUILD:
```diff
@@ -1,33 +1,43 @@
     resource_files = glob(["src/main/res/**"]),
     resources = main_resources,
-    tags = ["maven_coordinates=com.intuit.playerui:android:aar:%s" % VERSION],
     visibility = ["//visibility:public"],
```
One last thing - since this is now an intermediate target, can we remove the public visibility here to ensure usage of this flows through the end target?

sugarmanz reviewed (dismissed):
Pending visibility change on intermediate `player_android` target.
